.buildkite/hooks/pre-exit

```diff
@@ -10,7 +10,7 @@ if [[ "$BUILDKITE_PIPELINE_SLUG" == "elastic-agent" && "$BUILDKITE_STEP_KEY" ==
 
   # Perform cleanup of integration tests resources
   echo "--- Cleaning up integration test resources"
-  TEST_INTEG_AUTH_ESS_REGION=us-east-1 SNAPSHOT=true mage integration:clean
+  SNAPSHOT=true mage integration:clean
```

Comment:
I assume the serverless errors are originating here. The `.integration-cache` needs to have some kind of data to tell us if the current deployment is serverless or stateful, and then we can create the appropriate provisioner.

Reply:
I don't believe so. It is originating here - https://github.com/blakerouse/elastic-agent/blob/35dbbdea9b7bf9693864564e58b9e590e68104f3/pkg/testing/ess/serverless.go#L209

```
>>> (linux-amd64-ubuntu-2204) Failed for instance linux-amd64-ubuntu-2204 (@ 34.27.26.94): failed to check for serverless 8.12.0-SNAPSHOT to be ready: error waiting for ES to become available: error waiting for ES to become healthy: got context done; Last HTTP Error: unexpected status code 401 in request to https://snapshot-cae896.es.us-east-1.aws.elastic.cloud, body: {"error":{"root_cause":[{"type":"security_exception","reason":"unable to authenticate user [] for REST request [/]","header":{"WWW-Authenticate":["Bearer realm=\"security\"","ApiKey","Basic realm=\"security\" charset=\"UTF-8\""]}}],"type":"security_exception","reason":"unable to authenticate user [] for REST request [/]","header":{"WWW-Authenticate":["Bearer realm=\"security\"","ApiKey","Basic realm=\"security\" charset=\"UTF-8\""]}},"status":401}
```

I think I have identified the problem though, when that is called in the code now the username/password is not provided so it cannot actually authenticate.

I was thinking it was an issue with the API key and it is not.